lib/output.js

```diff
@@ -117,15 +117,37 @@ function OutputStream(options) {
         function quote_double() {
             return '"' + str.replace(/\x22/g, '\\"') + '"';
         }
+        function quote_template() {
+            if (!options.ascii_only) {
+                str = str.replace(/\\(r|n|u2028|u2029)/g, function(s, c) {
+                    switch(c) {
+                        case "n": return "\n";
+                        case "r": return "\r";
+                        case "u2028": return "\u2028";
+                        case "u2029": return "\u2029";
+                    }
+                    return "\\" + c;
+                });
+            }
+            return '`' + str.replace(/`/g, '\\`') + '`';
+        }
         if (options.ascii_only) str = to_ascii(str);
         switch (options.quote_style) {
           case 1:
+            if (quote == "`" && sq > 0)
+                return quote_template();
```

Comment:
Condition looks wrong. What do the number of single quotes have to do with outputting a template string for `quote_style=1`? Ditto for case 2 below and dq.

Also: need tests with one or more embedded back quotes ``` \` ``` within template strings with various quote_styles.